### DISCUSSION TITLE
how can i display label of detection in another language ?

### DISCUSSION THREAD

**Discussion by 7assanx7** [roboflow/supervision#38] [Q&A]
![object+detection](https://user-images.githubusercontent.com/118318017/223698865-ae5f0fdf-1658-40c8-94dc-a27319a2fd0c.png)
i want to display prediction of bicycle in another language like arabic language so how can i do that in this code :

import cv2
import argparse
import pyttsx3

from ultralytics import YOLO
import supervision as sv

import numpy as np

def parse_arguments() -> argparse.Namespace:
    parser = argparse.ArgumentParser(description="YOLOv8 live")
    parser.add_argument(
        "--webcam-resolution",
        default=[1280, 720],
        nargs=2,
        type=int
    )
    args = parser.parse_args()
    return args

def main():
    args = parse_arguments()
    frame_width, frame_height = args.webcam_resolution
    cap = cv2.VideoCapture(0)
    cap.set(cv2.CAP_PROP_FRAME_WIDTH, frame_width)
    cap.set(cv2.CAP_PROP_FRAME_HEIGHT, frame_height)

    model = YOLO("runs/detect/train29/weights/best.pt")

    box_annotator = sv.BoxAnnotator(
        thickness=2,
        text_thickness=2,
        text_scale=1
    )

    while True:
        ret, frame = cap.read()
        font = cv2.FONT_HERSHEY_SIMPLEX

        result = model(frame, agnostic_nms=True)[0]
        detections = sv.Detections.from_yolov8(result)
        labels = [
            f"{model.model.names[class_id]} {confidence:0.75f}"
            for _, confidence, class_id, _
            in detections
        ]
        frame = box_annotator.annotate(
            scene=frame,
            detections=detections,
            labels=labels
        )

        cv2.imshow("yolov8", frame)

        if (cv2.waitKey(30) == 27):
            break

if __name__ == "__main__":
    main()
*****************************************************************
in specific more i want if yolo detect class 0 or bicycle display text "the label in any language " on webcam frame

**Comment by SkalskiP** [comment]
Hi, @7assanx7 👋🏻! Your labels are resolved here: 

```
labels = [
    f"{model.model.names[class_id]} {confidence:0.75f}"
    for _, confidence, class_id, _
    in detections
]
```

If you want to display a different name you'll need to change this part `f"{model.model.names[class_id]} {confidence:0.75f}"`.

**Reply by 7assanx7** [reply]
What should i do in this part to display same result in different language?

**Reply by SkalskiP** [reply]
How about that? In general `if` class_id is `0` it will display `ARABLIC_LABEL` and `ENGLISH_LABEL` in other cases. IS that what you are looking for? 

```python
labels = [
    "ENGLISH_LABEL" if class_id != 0 else "ARABLIC_LABEL"
    for _, confidence, class_id, _
    in detections
]
```

**Reply by 7assanx7** [reply]
Thank you very much it worked
last thing do you know how to fix the quistion marks when i used arabic language
![Screenshot 2023-03-13 141545](https://user-images.githubusercontent.com/118318017/224699704-3c52f299-dae0-4a92-9096-99649350bbe1.png)

**Reply by SkalskiP** [reply]
Oh my... I did some research, and it looks like I have bad news for you. I'm really sorry, but OpenCV (the library we sue for text rendering in Supervision) does not support the Arabic alphabet :/ We can add an alternative rendering engine, but that will not happen soon. I'm super sorry.

**Reply by 7assanx7** [reply]
> Oh my... I did some research, and it looks like I have bad news for you. I'm really sorry, but OpenCV (the library we sue for text rendering in Supervision) does not support the Arabic alphabet :/ We can add an alternative rendering engine, but that will not happen soon. I'm super sorry.

I understand, no problem. thanks for your help

**Comment by Davisonyeas** [comment]
How can I display the image of only 1 class that is detected?
